lib/speed.js

```diff
@@ -1,37 +1,127 @@
 'use strict';
 
-const rpad = (s, char, count) => (s + char.repeat(count - s.length));
-const lpad = (s, char, count) => (char.repeat(count - s.length) + s);
+const fs = require('fs');
+const { fork } = require('child_process');
+const makeResults = require('./speed/make-results.js');
+const PATH_GET_TIME = __dirname + '/speed/get-time.js';
+const GET_NODE_PATH = __dirname + '/speed/get-node-path.sh';
 
-const speed = (
-  caption,
-  count,
-  tests
+const getNodePath = (
+  // Function which get path to node by version
+  version // String, versions of node, example: 9.4.0, 8.0, 7
+  // Returns: String, path to node
 ) => {
-  const times = tests.map((fn) => {
-    const result = [];
-    const begin = process.hrtime();
-    let i;
-    for (i = 0; i < count; i++) result.push(fn());
-    const end = process.hrtime(begin);
-    const time = end[0] * 1e9 + end[1];
-    const name = rpad(fn.name, '.', 25);
-    return { name, time };
-  });
-  console.log();
-  const top = times.sort((t1, t2) => (t1.time - t2.time));
-  const best = top[0].time;
-  const relative = (time) => (time * 100 / best);
-  top.forEach((test) => {
-    test.percent = Math.round(
-      Math.round(relative(test.time) * 100) / 100
-    ) - 100;
-    const time = lpad(test.time.toString(), '.', 15);
-    const percent = lpad((
-      test.percent === 0 ? 'min' : '+' + test.percent + '%'
-    ), '.', 10);
-    console.log(test.name + time + percent);
+  const execSync = require('child_process').execSync;
+  let pathToNode = null;
+  if (process.env.NVM_DIR) {
+    pathToNode = execSync(['sh', GET_NODE_PATH, version, 'nvm'].join(' '))
+      .toString().trim();
+    if (fs.existsSync(pathToNode)) return pathToNode;
+  }
+  if (process.env.NVS_HOME) {
+    pathToNode = execSync(['sh', GET_NODE_PATH, version, 'nvs'].join(' '))
+      .toString().trim();
+    if (fs.existsSync(pathToNode)) return pathToNode;
+  }
+  throw new Error('Node: ' + version + ' not found');
+};
+
+const getFullVersion = (
+  // Function which get full node version from node path
+  pathToNode // String, path to node
+  // Returns: String, full node version
+) => {
+  const path = pathToNode.split('/');
+  const version = path[path.length - 3];
+  return version[0] === 'v' ? version.substring(1, version.length) : version;
+};
+
+const makeRequests = (
+  // Function which create requests to child processes
```

Comment:
Typo: you should use `creates` instead of `create`, although I'd change it to `prepares`.